registry/registry_ui_test.go

```diff
@@ -267,3 +279,311 @@ func TestPopulateRegistrationFields(t *testing.T) {
+		}()
+
+		getInsts, intErr, extErr := getCachedInstitutions()
+		require.NoError(t, intErr)
+		require.NoError(t, extErr)
+		assert.Equal(t, mockInsts, getInsts)
+
+		func() {
+			institutionsCacheMutex.Lock()
+			defer institutionsCacheMutex.Unlock()
+			institutionsCache.DeleteAll()
+		}()
+	})
+
+	t.Run("cache-miss-with-success-fetch", func(t *testing.T) {
+		viper.Reset()
+		logrus.SetLevel(logrus.InfoLevel)
+		hook := test.NewGlobal()
+		defer hook.Reset()
+
+		// This is dangerous as we rely on external API to decide if the test succeeds,
+		// but this is the one way to test with our custom http client
+		viper.Set("Registry.InstitutionsUrl", "https://topology.opensciencegrid.org/institution_ids")
+		institutionsCache = ttlcache.New[string, []Institution]()
+
+		getInsts, intErr, extErr := getCachedInstitutions()
+		require.NoError(t, intErr)
+		require.NoError(t, extErr)
+		assert.Greater(t, len(getInsts), 0)
+		assert.Equal(t, 1, len(hook.Entries))
+		assert.Contains(t, hook.LastEntry().Message, "Cache miss for institutions TTL cache")
+
+		func() {
+			institutionsCacheMutex.Lock()
+			defer institutionsCacheMutex.Unlock()
+			institutionsCache.DeleteAll()
+		}()
+	})
+
+	t.Run("cache-hit-with-two-success-fetch", func(t *testing.T) {
+		viper.Reset()
+		logrus.SetLevel(logrus.InfoLevel)
+		hook := test.NewGlobal()
+		defer hook.Reset()
+
+		// This is dangerous as we rely on external API to decide if the test succeeds,
+		// but this is the one way to test with our custom http client
+		viper.Set("Registry.InstitutionsUrl", "https://topology.opensciencegrid.org/institution_ids")
+		institutionsCache = ttlcache.New[string, []Institution]()
+
+		getInsts, intErr, extErr := getCachedInstitutions()
+		require.NoError(t, intErr)
+		require.NoError(t, extErr)
+		assert.Greater(t, len(getInsts), 0)
+		assert.Equal(t, 1, len(hook.Entries))
+		assert.Contains(t, hook.LastEntry().Message, "Cache miss for institutions TTL cache")
+
+		hook.Reset()
+
+		getInsts2, intErr, extErr := getCachedInstitutions()
+		require.NoError(t, intErr)
+		require.NoError(t, extErr)
+		assert.Greater(t, len(getInsts2), 0)
+		assert.Equal(t, getInsts, getInsts2)
+		// No cache miss
+		assert.Equal(t, 0, len(hook.Entries))
+
+		func() {
+			institutionsCacheMutex.Lock()
+			defer institutionsCacheMutex.Unlock()
+			institutionsCache.DeleteAll()
+		}()
+	})
+}
+
+func TestCheckUniqueInstitutions(t *testing.T) {
+	t.Run("empty-gives-true", func(t *testing.T) {
+		unique := checkUniqueInstitutions([]Institution{})
+		assert.True(t, unique)
+	})
+
+	t.Run("unique-gives-true", func(t *testing.T) {
+		unique := checkUniqueInstitutions([]Institution{{ID: "1"}, {ID: "2"}})
+		assert.True(t, unique)
+	})
+
+	t.Run("duplicated-gives-false", func(t *testing.T) {
+		unique := checkUniqueInstitutions([]Institution{{ID: "1"}, {ID: "1"}})
+		assert.False(t, unique)
+	})
+
+	t.Run("large-entries", func(t *testing.T) {
```

Comment:
What does this test check that the previous doesn't?

Reply:
The idea was to add some relatively "large" number of entries to make sure we are not just reading the first couple of entries to check for duplicates